### DISCUSSION TITLE
Inconsistent behavior for `dataset.num_classes` and `dataset.num_features`

### DISCUSSION THREAD

**Discussion by EdisonLeeeee** [pyg-team/pytorch_geometric#5273] [General]
Hi, I am using PyG datasets with my custom transform function, but I found there is an inconsistent behavior between `dataset.num_classes` and `dataset.num_features`. The minimal code to reproduce my scenario is as follows:
```python
import torch
from torch_geometric.datasets import Planetoid

def my_transform(data):
    data.x = torch.randn(100, 10) # num_features changed here
    data.y = torch.arange(100) # num_classes changed here
    return data

dataset = Planetoid('~/data/Cora', 'Cora')
print(f"num_features: {dataset.num_features}, num_classes: {dataset.num_classes}")
# num_features: 1433, num_classes: 7

dataset = Planetoid('~/data/Cora', 'Cora', transform=my_transform)
print(f"num_features: {dataset.num_features}, num_classes: {dataset.num_classes}")
# num_features: 10, num_classes: 7
```

That is, `dataset.num_features` returns the results from the transformed dataset while `dataset.num_classes` does not. I am not sure if this is intended behavior or a potential bug.

**Comment by rusty1s** [comment]
Thanks for reporting. I just fixed that here: https://github.com/pyg-team/pytorch_geometric/pull/5274

**Reply by EdisonLeeeee** [reply]
Great, thanks!
